### DISCUSSION TITLE
Ask us anything about the company

### DISCUSSION THREAD

**Discussion by phazonoverload** [directus/directus#17034] [Team Posts]
Got a question for the Directus founders? I'll be sitting down with our founders @benhaynes and @rijkvanzanten for a Q&A in a couple of weeks time, so tell us what you want to know! 

Given that we are an open source project with a company behind it, I'm particularly interested in questions about the **company that maintains the project** such as direction, concerns, decision-making, or anything else really. 

Reply here or [use this form](https://share.hsforms.com/1fp5LfBjRR6ereNmIworpJQc8497) to submit questions. This isn't a one-off either and we plan to do this every few months. So the form and comments thread will remain open. 😇 🐰 💜

**Comment by navanshu** [comment]
How did you develop Flows for Directus, what inspired the syntax.

**Reply by paescuj** [reply]
Interesting question! I think it has been inspired by [BPMN](https://en.m.wikipedia.org/wiki/Business_Process_Model_and_Notation) but of course the founders will be able to give you the exact answer 😃

**Comment by danielband** [comment]
Hi, I am following Directus since 2018. If I remember correctly, this was when v7 came out. And that version was a true nightmare, throwing errors all over the place after starting the container. Obviously Directus has come a long way since then, and you guys are doing a great job. Improving year after year. You even implemented OIDC as a free feature, when so many players are claiming it to be "enterprise". So first off, I want to thank the whole team for the work you did and the you will put in. Here are my questions:

1. How did you persevere through initial times (basically until v9, which I consider the first usable product) and what continues to drive you?
2. As of now, there are many viable use cases for Directus, since the core is very open. Feature requests are all over the place, may it be dev xp, basic website building or more complex data processing. You wont be able to please them all. In which direction do you see Directus going? How do you balance the requests from (big) bakers vs community? 
4. Is there anything regarding the project you guys are concerned about?

Cheers,
Daniel

**Comment by unleashit** [comment]
Have really enjoyed discovering and working with Directus over the last month. It had been several years since I explored the space, so I spent some time comparing the (open source and self hosted) options, and the build quality of Directus stands out. Especially when it comes to the admin, which is really what these tools are for. Still, there are a few areas where I think think it could improve, especially to support more use cases. From my perspective these would be:

- Simple way in admin to rename DB table and field names. I can't tell you how many times I used singular, when I really wanted plural, etc. but then gave up because I was too lazy to rename in 6-8 tables.

- A simple programmatic way to define schema in code. This is where for example Payload CMS shines, but their backend well... if we're allowed to honest... kind of sucks and hard to imagine end users enjoying. They also don't have near the media features of Directus, and I'm sure there's more.

- Ability to attach to an existing Node or Express server. The local/extensions API is well thought out, but for a simple use case this would be great

- Option to generate thumbnails in advance as I believe previous Directus versions. This would allow hosting of files/images on external storage without the need to hit Directus (Diretus API could for example be consumed only at build time or from CI/CD).  Not everyone needs this, but it's really a good option for sites/apps that are either small or who don't get a lot of media updates. I believe there are acceptable solutions to file security for those who need it, and it could be opt in.

- The ability to deploy serverlessly 

- The ability to store files in a real (non-virtual) folder structure and let you move between storage locations in the GUI (at least unlock the field to make manually moving easier). The former seems to have different opinions, but there are plenty of situations when you end up needing to manage a large media collection outside of the normal machine managed situation. For similar reasons, it would also be nice include the human readable title in the file names.

This isn't meant to be criticism, I've found workarounds for most of my problems so far. But I do wonder if the cloud business model might be holding back some alternative deployment use cases. Obviously, you have to and deserve to make money. And thus whilie this has read so far like a feature request list, it does actually boils down to a company question. What do you think about a dual business model where you keep the cloud, but also offer a multi-site license for extended features on the open source self-hosted option?

The multi-site I think would stand out from most other offerings and probably convert a lot of otherwise non-paying developers who have projects that make no/little money. Obviously this isn't exactly a gold mine, but I don't think you would lose much from the cloud offering (I personally would far rather recommend that to a paying client so I don't have to maintain). But it could at the very least pay for focusing on some self-hosted specific concerns and expand the user base (leading to more sales).

Just my 2cents. I'm a fan in any case!

**Comment by AaronBeaudoin** [comment]
### What is in store for Directus in 2023?

Yes, I know the team has more recently been pretty tight lipped about the "timeline" for new features since a sponsorship is required, but we've also sort of ended up with a scenario where new features seem to be releasing in only a very slow trickle. **This is not to minimize the huge amount of work I'm sure the team is pouring into Directus right now.**

It is worth pointing out, however, that a large portion of the features which have been on the roadmap for a while still have yet to materialize. The Wayback Machine is a [wonderful thing](https://web.archive.org/web/20210925203619/https://directus.io/road-map/). As a software developer, I get it. Rarely are we able to get features rolled out in the timeframe we originally expect.

What I suggest is to do a monthly update/newsletter like [this](https://svelte.dev/blog). It's a really great way to build trust with your community that progress is actually being made towards goals. Just a little bit of insight into what the team is working on each month would go a **long** way towards making Directus feel like it isn't fizzling out. If new features are getting pushed back because the team is fixing a ton of bugs, you can just say that! "This month we rolled out fixed for over 45 bugs" or whatever. Transparency is critical.

**Reply by phazonoverload** [reply]
I think the perceived change in the degree of transparency wasn't a choice, but a byproduct of how much time the team has had while still actively developing the product and working to answer questions folks have. But I'm here now running Developer Relations, and increasing transparency is one of the very top things on my list. This exercise in sitting down with Ben and Rijk is a small part of that, but I just want you to know it's at the very forefront of my mind along with the rest of the team. 

I can't promise that we're going to be more forthcoming about when things are going to drop - I agree with the direction that a promise not made is a promise not broken - but in general you should expect more from us around comms (and asking for input from the community).

**Reply by rijkvanzanten** [reply]
> but a byproduct of how much time the team has had while still actively developing the product and working to answer questions folks have

I second this! 🙋🏻‍♂️

**Comment by Vishal9350108** [comment]
Nice

**Comment by lexoyo** [comment]
Hi

> Context: I'm developing [silex open source website builder](https://www.silex.me/) for 15 years now, and we are using it at [my company](https://internet2000.net/) along with directus and [11ty](https://www.11ty.dev/) to create green websites for customers

My question is: if we decided to offer a SaaS version of this "aggregate" of Silex+Directus+11ty (a great alternative to webflow), would Directus be happy about it? It would all be open source but maybe competing with directus cloud? Would you rather keep the hosting of Directus but it would make it more complex to setup for the end user?

Thank you, vive directus \o/

**Reply by Vishal9350108** [reply]
Nice bro ❤️🤩🥰

**Comment by phazonoverload** [comment]
Oh just so you all know, I recorded a video last week that I'm just editing. But this will be a regular thing so keep questions coming!

**Reply by Vishal9350108** [reply]
Nice

**Comment by Vishal9350108** [comment]
Nice ❤️🤩

**Comment by MentalGear** [comment]
First off:  You have build an amazing product for the open-source community - thank you for that !

1. Cloud is an amazing option, but I which there was a tier between Standard and Pro that would allow for more customization of endpoints, settings and custom code while not requiring the pro amount of storage or bandwidth. Use-case: Content Backend for Apps that are otherwise hosted statically and only rebuild on content change. (low write, high read)

2. Since your roadmap contains NoSQL database support for the near feature, do you also plan on supporting Graph DBs - which make reasoning and structuring data much more idiomatic and are a huge advantage to use as source for ML models - like SurrealDB?
https://github.com/surrealdb/surrealdb

**Comment by phazonoverload** [comment]
I wanted to share with you all the first Community Question Time, featuring questions we got here and elsewhere. https://directus.io/blog/community-question-time-0/

We cover:
- What is the value of community?
- What have been some of our mistakes?
- How do we balance business and community?
- Why are we slower at responding to Discord questions? 
- How do we decide what to work on? 
- How are you feeling about our first board meeting? 
- What vision did you sell to VCs? 
- How else might we make money? 
- How can we keep calling ourselves open source? 
- What are you excited about in 2023? 
- What are you worried about in 2023?

We'll do these every few months, and I know we didn't get to every question asked but it was already a 20 min video 😅 But we'll also be doing a regular series with @rijkvanzanten specifically on the technical questions you have (as opposed to Community Question Time, which is more about the organization). 

✌🏻

**Reply by Vishal9350108** [reply]
Nice

**Comment by unleashit** [comment]
Great video. Even if my personal question/suggestion wasn't directly addressed, the broader issues of how to balance the community (and growing the user base) vs commercial needs were . The takeaway for me was the team seems both competent and that they genuinely care about both. I still think there could be gold in that mine of increasing self hosted use cases (more deployment options open up whole new categories of users, which could translate to $), but it's obviously a risk and you guys are definitely the best ones figure out priorities.

**Reply by phazonoverload** [reply]
Oh yeah I’ve still got questions saved for the next one. We had a lot. Thanks so much for the kind  words

**Reply by Vishal9350108** [reply]
***Ni****ce***_

**Comment by koksikus** [comment]
Hi All,
I'm wondering how you determine the severity of the bug. Which function is basic and should work because it proves the quality of the application, i.e. basic specific functions always work (they will be fixed in the next release). 
I understand new functionalities and here I agree that if someone wants to speed up something, let them pay. Although there is one more problem here, that I paid for this functionality, it worked for a while and stopped again, and again you have to pay to start working, despite the fact that someone broke something in another place in the code. 
I mean that I generally like this application and I would like to show it to my boss so that an idea of ​​maintenance of tools could be implemented, i.e. a base with tools in Directus and it would be good if it showed which one is finishing the review, etc. And here I would see your help for which remuneration is due. 
But now how to conduct a presentation, since there are places where either the data is not displayed correctly or the rules cannot be fully applied (11058; maybe 17242 etc).

**Reply by phazonoverload** [reply]
I’m not quite following. I see the clear question at the top about how we rank severity, but I’m not sure if there’s another question in the rest of the message? 

Can you clarify? I want to make sure that we have time for all questions - so I need to make sure I understand them :)

**Reply by koksikus** [reply]
Main question is what features will be fixed immediately. 
I've been with you for a long time and I noticed that since November the number of releases and fixes has decreased, although the application is still not perfect.  It used to take a few days to confirm that something is a bug, now after 3 weeks we get information - pay and we will speed up to recognize something as a bug. 
It is probably difficult to answer, but have you identified the functions that must always be operational in order to show the quality of the application and stand out from others on the market.

**Reply by phazonoverload** [reply]
Right I got it! It was more info around severity ranking, which is something I can certainly spend time asking!

**Comment by rodrigoinhaia** [comment]
Hello! I would like to know if there is any idea of ​​making something similar to a GED (electronic document generator)? Like I can generate a word or pdf file based on a template, and the system fills in fields based on variables from my system or database? example:
"Address declaration of {{Client_Name}} for due purposes.

 {{street}}, no. {{number}}, {{client_city}}..."

I think I managed to explain it.
Do you have any predictions for this?

**Reply by nalognl** [reply]
Documets are always too custom for every company. We did such functionality for us via flows. 
We have button which make call te internal api server which get record id, get data from directus, generate document and upload generated filed to documents in directus and write comment. 

So you can easily create generator. Platform can live without it such as it can be easily achived via flows. But this is my opinion.

**Reply by rodrigoinhaia** [reply]
Hello, I tried to do it myself but I didn't get the result, could you help me?
rodrigo@sd.inf.br is my email if you want to send a print.
I can manipulate the data but I'm not able to pass it to an HTML document and open the print page, but you mentioned that it generated the file so could you explain to me better how you did it?

**Comment by nalognl** [comment]
Do you use Directus as CRM for your main company ? We do :)

**Reply by Syed-Sheharyar** [reply]
# How many use cases Directus has always amazed me.
